drv/mock-gimlet-seq-server/src/main.rs

```diff
@@ -9,43 +9,60 @@
 
 use drv_gimlet_seq_api::{PowerState, SeqError};
 use idol_runtime::RequestError;
-use userlib::RecvMessage;
+use task_jefe_api::Jefe;
+use userlib::{FromPrimitive, RecvMessage, UnwrapLite};
+
+userlib::task_slot!(JEFE, jefe);
 
 #[export_name = "main"]
 fn main() -> ! {
     let mut buffer = [0; idl::INCOMING_SIZE];
     let mut server = ServerImpl {
-        state: PowerState::A2,
+        jefe: Jefe::from(JEFE.get_task_id()),
     };
 
+    server.set_state_impl(PowerState::A2);
+
     loop {
         idol_runtime::dispatch(&mut buffer, &mut server);
     }
 }
 
 struct ServerImpl {
-    state: PowerState,
+    jefe: Jefe,
+}
+
+impl ServerImpl {
+    fn get_state_impl(&self) -> PowerState {
+        // Only we should be setting the state, and we set it to A2 on startup;
+        // this conversion should never fail.
+        PowerState::from_u32(self.jefe.get_state()).unwrap_lite()
+    }
+
+    fn set_state_impl(&self, state: PowerState) {
```

Comment:
Nit: if we always want to set the state on startup, should we make `ServerImpl::new(jefe: Jefe)` do it?

Reply:
I actually had it that way originally, then thought it was weird for `::new()` to have a side effect like that. Does that seem okay to you, or maybe there's a better name than `::new()`?

Reply:
Maybe `::init()` instead?  I normally associate that with more side effects.

Reply:
Sounds good; changed in c55fdaf